rules/python/lang/eval_using_user_input.yml

```diff
@@ -0,0 +1,76 @@
+imports:
+  - python_shared_common_external_input
+patterns:
+  - pattern: eval($<...>$<EXTERNAL_INPUT>$<...>)
+    filters:
+      - variable: EXTERNAL_INPUT
+        detection: python_shared_common_external_input
+        scope: result
+  - pattern: $<LITERAL_EVAL>($<...>$<EXTERNAL_INPUT>$<...>)
+    filters:
+      - variable: LITERAL_EVAL
+        detection: python_lang_eval_using_user_input_literal_eval
+        scope: result
+      - variable: EXTERNAL_INPUT
+        detection: python_shared_common_external_input
+        scope: result
+  - pattern: $<SUBINTERPRETERS>.run_string($<_ID>, $<...>$<EXTERNAL_INPUT>$<...>)
+    filters:
+      - variable: SUBINTERPRETERS
+        detection: python_lang_eval_using_user_input_subinterpreters
+      - variable: EXTERNAL_INPUT
+        detection: python_shared_common_external_input
+        scope: result
+auxiliary:
+  - id: python_lang_eval_using_user_input_literal_eval
+    patterns:
+      - pattern: $<AST_LITERAL_EVAL>
+        filters:
+          - variable: AST_LITERAL_EVAL
+            regex: \A(ast\.)?literal_eval\z
```

Comment:
Maybe this is overkill, but I think this would cover all cases:
```suggestion
      - pattern: $<AST_MODULE>.literal_eval
        filters:
          - variable: AST_MODULE
            detection: python_lang_eval_using_user_input_ast_module
            scope: cursor
      - from ast import $<!>literal_eval
      - from ast import literal_eval as $<!>$<_>
  - id: python_lang_eval_using_user_input_ast_module
    patterns:
      - import $<!>ast
      - import ast as $<!>$<_>
```